design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,278 @@
+    sku: String!
+    quantity: Float!
+    parent_sku: String,
+    note: String,
+    # see https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md
+    selected_options: [String!]
+    # see https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/add-items-to-cart-single-mutation.md
+    entered_options: [EnteredOptionInput!]
+}
+
+input UpdateGiftRegistryInput {
+    event_name: String
+    type_id: String
+    message: String
+    privacy_settings: GiftRegistryPrivacySettings
+    status: GiftRegistryStatus
+    registrants: [UpdateGiftRegistryRegistrantInput!] @doc(description: "As a result of the update, only the records of provided registrants will be affected. If the registrant is missing in the request, its attributes will not be changed")
+    shipping_address: GiftRegistryShippingAddressInput
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput!] @doc(description: "As a result of the update, only the values of provided attributes will be affected. If the attribute is missing in the request, its value will not be changed")
+}
+
+input CreateGiftRegistryInput {
+    event_name: String!
+    type_id: String!
+    message: String!
+    privacy_settings: GiftRegistryPrivacySettings!
+    status: GiftRegistryStatus!
+    registrants: [CreateGiftRegistryRegistrantInput!]!
+    shipping_address: GiftRegistryShippingAddressInput
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput]
+}
+
+type GiftRegistryShippingAddressInput @doc(description: "Either address data or address ID should be provided") {
+    address_data: CustomerAddressInput
+    address_id: Int
+}
+
+type UpdateGiftRegistryRegistrantInput {
+    id: ID!
+    first_name: String
+    last_name: String
+    email: String
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput] @doc(description: "As a result of the update, only the values of provided attributes will be affected. If the attribute is missing in the request, its value will not be changed")
+}
+
+type CreateGiftRegistryRegistrantInput {
+    first_name: String!
+    last_name: String!
+    email: String!
+    dynamic_attributes: [GiftRegistryDynamicAttributeInput]
+}
+
+type GiftRegistryDynamicAttributeInput {
+    code: ID!
+    value: String!
+}
+
+type CreateGiftRegistryOutput {
+    gift_registry: GiftRegistry
+}
+
+type UpdateGiftRegistryOutput {
+    gift_registry: GiftRegistry
+}
+
+type AddProductsToGiftRegistryOutput {
+    gift_registry: GiftRegistry
+}
+
+type GiftRegistryType {
+    id: ID!
+    label: String!
+    dynamic_attributes_metadata: [GiftRegistryDynamicAttributeMetadataInterface]
+}
+
+interface GiftRegistryDynamicAttributeMetadataInterface {
+    code: ID!
+    input_type: String!
+    attribute_group: String!
+    label: String!
+    is_required: Boolean!
+    sort_order: Int
+}
+
+type GiftRegistryTextAttributeMetadata implements GiftRegistryDynamicAttributeMetadataInterface {
```

Comment:
![image](https://user-images.githubusercontent.com/5233399/83556934-61dc3700-a4d6-11ea-94a9-cc52fdb5f207.png)

http://spec.graphql.org/June2018/#sec-Objects

Reply:
That is correct, however we have custom schema stitching in Magento GraphQL webonyx implementation, which copies all fields from interfaces to the implementation, before producing the final merged schema. This allows us to avoid duplication. So we also follow the same style in designs, then implementors can just take schema from the design document and it should work.

Reply:
Ah gotcha - @danielrenaud had to explain this to me a few hours ago.